src/main/java/com/vmware/vipclient/i18n/base/HttpRequester.java

```diff
@@ -269,8 +278,17 @@ public void checkServerTrusted(X509Certificate[] cert,
 	}
 
 	public boolean isConnected() {
-		String ipAddress = (vipHostName.contains(":")) ? vipHostName.split(":")[0]
-				: vipHostName;
+		String ipAddress = (vipHostName.contains(":")) ? vipHostName.split(":")[0] : vipHostName;
 		return HttpRequester.ping(ipAddress);
 	}
+
+	private void addHeaderParams(HttpURLConnection connection) {
+		if (null == customizedHeaderParams) {
```

Comment:
it's also better to check if connection is null. so please change it to this?
if (null == customizedHeaderParams || connection == null)